requests_aws4auth/aws4auth.py

```diff
@@ -577,13 +578,20 @@ def get_sig_string(req, cano_req, scope):
         """
         Generate the AWS4 auth string to sign for the request.
 
+        Checks to see if date has been passed as x-amz-date or date, if
+        date then parse the date from an RFC 7231 string.
```

Comment:
if only date is present.

email.utils doesn't seems to mention RFC 7231 so are you sure it does?